File: fuel-distribution-system/allocation-service/order-service/src/main/java/com/gihan/orderservice/service/OrderService.java
package com.gihan.orderservice.service;

import com.gihan.orderservice.model.Order;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface OrderService {
    List<Order> getOrders();
    Order createOrder(Order order);
    ResponseEntity<Order> getOrderById(int id);
    ResponseEntity<Order> updateOrder(int id, Order orderDetails);
    ResponseEntity<Map<String, Boolean>> deleteOrder(int id);
}

File: front-end/order-frontend/src/app/order/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http' //imported
import { Observable } from 'rxjs'; //imported
import { Order } from "../order/order";

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  private baseUrl = "http://localhost:8080/orders";
  private creatOrderUrl = "http://localhost:8080/order";
  private getOrderByIdUrl = "http://localhost:8080/order"; // http://localhost:8080/order/2
  private deleteOrderByIdUrl = "http://localhost:8080/order"; // http://localhost:8080/order/2

  constructor(private httpClient: HttpClient) {} // imported and injected http client

  getOrders():Observable<Order[]>{
    return this.httpClient.get<Order[]>(this.baseUrl);
  }

  creatOrder(order:Order):Observable<Object>{
    return this.httpClient.post(this.creatOrderUrl,order);
  }

  getOrderById(id:number):Observable<Order>{
    return this.httpClient.get<Order>(`${this.getOrderByIdUrl}/${id}`);
  }

  updateOrder(order:Order,id:number):Observable<Order>{
    return this.httpClient.put<Order>(`${this.getOrderByIdUrl}/${id}`,order);
  }

  deleteOrder(id:number):Observable<Object>{
    return this.httpClient.delete(`${this.deleteOrderByIdUrl}/${id}`);
  }
}
